packages/cli/src/commands/styles/utils/publish.main.ts

```diff
@@ -21,39 +22,71 @@ export const generateToken = (name: string) => {
 export const filterFunctions: NodeFilter[] = [];
 export const designTokenFunctions: DesignTokenFilter[] = [];
 
-const colorDesignTokenizer: DesignTokenFilter = (node: NodeDocument): DesignToken => {
-  colorToHex(node.fills[0].color);
-  const colorToken: DesignToken = {
-    type: 'color' ,
-    name: node.name,
-    token: generateToken(node.name),
-    value: colorToHex(node.fills[0].color)
-  };
-  return colorToken;
-};
 
-const colorFilter: NodeFilter = (data: NodeDocument, designTokenFilterFn: DesignTokenFilter): DesignToken => {
-  if(data.type == 'RECTANGLE') {
-    data.type = 'color';
+export const processStyleNodes = (data: NodeDocument, type: string): DesignToken[] | DesignToken | undefined => {
+  if(!data) return undefined;
+  switch(type){
+    case 'FILL':
+      // simple
+      return colorDesignTokenizer(data);
+    case 'TEXT':
+      return typographyTokenizer(data);
+    case 'EFFECT':
+      return processElevationTokenizer(data);
+    default:
+      return undefined;
+  }
+}; 
+
+
+// const colorFilter: NodeFilter = (data: NodeDocument, designTokenFilterFn: DesignTokenFilter): DesignToken => {
+//   if(data.type == 'RECTANGLE') {
+//     data.type = 'color';
+//   }
+//   const colorToken = designTokenFilterFn(data);
+//   return colorToken;
+// };
+
+// const typographyFilter: NodeFilter = (data: NodeDocument, designTokenFilterFn: DesignTokenFilter): DesignToken => {
+//   if(data.type == 'RECTANGLE') {
+//     data.type = 'color';
+//   }
+//   const colorToken = designTokenFilterFn(data);
+//   return colorToken;
+// };
+
+// export const convertNodesToTokens = (nodes: NodeDocument[]): DesignToken[] => {
+//   const designTokens: DesignToken[] = [];
+//   nodes.forEach((node) => {
+//     filterFunctions.forEach((filterMethod, index) => {
+//       const dToken = filterMethod(node, designTokenFunctions[index]);
+//       designTokens.push(dToken);
+//     });
+//   });
+//   return designTokens;
+// };
+
+// // These methods have to be passed in order. Methods are run sequentially
+// filterFunctions.push(colorFilter,typographyFilter);
+// designTokenFunctions.push(colorDesignTokenizer);
+
+export const convertStyleNodesToTokens = (nodes: { [key: string]: NodeRoot }, styles: StyleMetadata[]) => {
+  let designTokens: DesignToken[] = [];
+  for(const index in styles){
+    const styleId = styles[index].node_id;
+    console.log(styleId);
+    const newTokens = processStyleNodes(nodes[styleId]?.document,styles[index].style_type);
+    if(newTokens){
+      if(Array.isArray(newTokens)){
+        designTokens = designTokens.concat(newTokens);
+      } else {
+        designTokens.push(newTokens);
+      }
+    }
```

Comment:
A typeguard that checks returns the correct type based on the properties. Its a typescript thing lol 
```
  const isArray = (dToken: DesignToken | DesignToken[]): dToken is DesignToken[] => {
    if(Array.isArray(dToken)) return true;
    return false;
  }; 

```

Just a small little refactor of the code. I'm also returning `designTokens[]`, since the return type for the function by default was void
  ```
const convertStyleNodesToTokens = (nodes: { [key: string]: NodeRoot }, styles: StyleMetadata[] | undefined) => {
    let designTokens: DesignToken[]  = [];
    styles?.forEach((style) => {
      const designToken = processStyleNodes(nodes[style.node_id].document, style.style_type);
      if(designToken) {
        isArray(designToken) ? designTokens = [...designToken] : designTokens.push(designToken);
      }
    });
    return designTokens;
  };
```